src/containers/lineup/hooks.ts

```diff
@@ -0,0 +1,74 @@
+import { useMemo } from 'react'
+import { useDispatch } from 'react-redux'
+import { makeGetLineupMetadatas } from 'store/lineup/selectors'
+import { getBuffering, getPlaying } from 'store/player/selectors'
+import { useSelector } from 'utils/reducer'
+import { LineupVariant } from 'containers/lineup/types'
+import { AppState } from 'store/types'
+import { LineupState } from 'models/common/Lineup'
+import { makeGetCurrent } from 'store/queue/selectors'
+import { isMobile } from 'utils/clientUtil'
+
+type LineupActions = any
+
+type UseLineupProps = {
+  getLineupSelector: (state: AppState) => LineupState<{}>
+  actions: LineupActions
+  variant?: LineupVariant
+  numPlaylistSkeletonRows?: number
+  scrollParent?: HTMLElement
+}
+
+/**
+ * Returns props for a Lineup component.
+ * Requires at least a selector and actions
+ * See example usage in `TrendingPlaylistPage`
+ * */
+export const useLineup = ({
```

Comment:
maybe we could call this useLineupProps since you still need to render separately? wdyt?

Reply:
Yeah I dig it